common/lib/src/realm_types.dart

```diff
@@ -21,32 +21,46 @@ import 'dart:typed_data';
 import 'package:objectid/objectid.dart';
 import 'package:sane_uuid/uuid.dart';
 
-/// All supported `Realm` property types.
-/// {@category Configuration}
+Type _typeOf<T>() => T;
+
+/// @nodoc
+class Mapping<T> {
+  const Mapping({this.indexable = false});
+
+  final bool indexable;
+
+  // Types
+  Type get type => T;
+  Type get nullableType => _typeOf<T?>();
+}
+
+const _intMapping = Mapping<int>(indexable: true);
+const _boolMapping = Mapping<bool>(indexable: true);
+
+/// @nodoc
 enum RealmPropertyType {
```

Comment:
RealmPropertyType is exposed and should have an API docs.

Reply:
The only reason it is exposed is because we use it in the generated classes, but I will add API docs back.

Reply:
Done